### ISSUE TITLE
Differences with the paper

### ISSUE THREAD

**Issue by lukeb88** [philipperemy/keras-tcn#38]
I noticed some differences in the implementation of the architecture compared to the original paper.

I've already come across another issue, where @philipperemy was answering: "As the author of the paper stated, this TCN architecture really aims to be a simple baseline that you can improve upon. So personally I think replicating the exact same results is not that important.".

I just wanted to know if there are any reasons behind these two implementation choices that I don't understand:

1) In the paper, they don't talk about stacking various dilated convolutions (parameter `nb_stacks`):
    I think I understood what you discussed [here](https://github.com/philipperemy/keras-tcn/issues/9), but how this changes instead of just increasing the number of dilations? 

2) In the single residual block (in `residual_block()`), why is there only one convolutional layer instead of 2 as in the paper?
    It's just a simplification?

**Comment by lingdoc** [comment]
Another question: the paper also mentions using Gradient Clipping for language modeling, but I don't see how that is implemented in this architecture. Is there another parameter setting for that?

**Comment by philipperemy** [comment]
@lingdoc this is in my TODO list. I'll dig into that when I have a bit of time. In ~1 week from now!

**Comment by lingdoc** [comment]
btw, great implementation! I'm comparing it with a BiLSTM-CNN variant on a text classification problem and getting similar results with half the runtime.

**Comment by philipperemy** [comment]
That's great! Thank you :)

**Comment by philipperemy** [comment]
Just for info, I'm working on a PR to make the implementation identical to the paper: https://github.com/philipperemy/keras-tcn/pull/42

**Comment by philipperemy** [comment]
Merged. Only weight norm is missing
https://github.com/philipperemy/keras-tcn/commit/141ef1c1fdfc2384aec5b24caa2e9624d486d9ef

**Comment by philipperemy** [comment]
@lukeb88 :

1. In the paper, they don't talk about stacking various dilated convolutions (parameter `nb_stacks`):
   I think I understood what you discussed [here](https://github.com/philipperemy/keras-tcn/issues/9), but how this changes instead of just increasing the number of dilations?

=> fixed I set the nb_stacks to 1 for all the examples. Results are the same.

how this changes instead of just increasing the number of dilations?

cf. https://arxiv.org/pdf/1609.03499.pdf

2. In the single residual block (in `residual_block()`), why is there only one convolutional layer instead of 2 as in the paper?
   It's just a simplification?

=> Was a simplification. But I fixed it!
